### ISSUE TITLE
Feature request :  Add a tab in Main Page for the Background Player

### ISSUE THREAD

**Issue by gituby** [TeamNewPipe/NewPipe#2477]
Hi NewPipe Team,

I wish to make a suggestion for a feature in the main page content. Could it be possible to add in the main content page, a tab for the background player to be able to access it easily (picture 1).

For this, in the parameters menu "Content of Main Page", when clicking on the "+" there could be a new option to add a tab to access directly the background player (picture 2). This will let the user decide to show this tab or not as he wish.

Thanks for reading

![Main Page - Background Player](https://user-images.githubusercontent.com/53324825/61981397-e0846000-aff9-11e9-8cbd-a133d2eae53c.jpg)

![Parameters - Content of Main¨Page](https://user-images.githubusercontent.com/53324825/61981415-eb3ef500-aff9-11e9-9d65-12d1738372cf.jpg)

**Comment by Stypox** [comment]
This is a great idea!

**Comment by Wilker-a** [comment]
i think that an always available button would be easier to navigate, as well as the plus button available in the same way in the playlist screen would be easier to navigate with.

![a mock-up of the button](https://user-images.githubusercontent.com/10854855/62001450-519e4300-b0e0-11e9-80b8-27579d36ff1e.jpg)

**Comment by Stypox** [comment]
@Wilker-a 
What if someone never uses the background player? We could add an option to enable/disable that floating-action-button (FAB), but adding too many options is never good.
Also, a custom background tab is less obstructive, since you can switch back and forth any tab seamlessly.
And adding both the button and the background tab is not great either, since there should only be one way to do something.

Material Design Guidelines suggests to use FABs only when the action they link to is very obvious and frequently used, and I don't think the button you suggest to add meets either of these requirements.

**Comment by Wilker-a** [comment]
i think the solution then should be to make it so the button could be shown only whenever there is a queue or when the media is playing at the background, as these seems to be the only use cases that would most likely happen for the use of the playlist button, and in these cases, it would be frequently used, because the notifications or going away from a video page is not always a reliable way to control media.

the tabs requires the user to go back once and maybe twice if they're browsing a video or searching for one, so i think that makes it a bit more difficult to access without obstructing the other tasks, and i think it would be useful only when the user starts up the app to get a playlist to play with a link for example.

one alternative would be to have the menu panel available at all times with a swipe and put the playlist button there, or put it in the 3 dots menu (together with the downloads and settings buttons and stuff).

also, making the player itself available at all times like a regular music player is not an option at the moment, because that would obstruct the comments section. unless the user exclusively uses SoundCloud or a service that only plays in the background.

**Comment by Wilker-a** [comment]
i think i forgot to mention this:

i think the plus button on the playlist could have secondary actions, like a paste button, that automatically takes the link from the clipboard to add the media or the playlist to the queue. whenever pressed.

![Screenshot from KeePass DX. This button expands whenwver clicked, and hidden whenever you scroll down. Shows up ](https://user-images.githubusercontent.com/10854855/62012184-793deb80-b172-11e9-8f8d-1c03967be43e.png)

..on a second thought, i think this would actually be the only function of the plus button (which in this case would be better as a button in the 3 dots menu, and with the chain icon for the link

**Comment by theScrabi** [comment]
The idea is good, i just dont like the fap button. It does not make sense in this context, as there is no list to which you can add items on this screen.

I'd rather add an additional menu option that comes with an image and a descriptuon.

**Comment by Wilker-a** [comment]
something like this?

![the button would add a single piece or entire playlists dependig on the url](https://user-images.githubusercontent.com/10854855/62016081-79a0ab80-b19f-11e9-9c6a-375fc6636d79.jpg)

**Comment by theScrabi** [comment]
No i ment the second screen that gituby posted.

**Comment by ronidee** [comment]
I agree with @Stypox. A FAB is bad UX and shouldn't be used in this context.

**Comment by avently** [comment]
I think it's a good idea. How about another one? Since we have mini player it can show a list button which shows (after click) a dialog with the play queue. It can be a dialog or a bottomSheet. More importantly by making this way you'll have easy access to the play queue. If you have a tab on main screen you would need to go back to the main screen which is not always useful, for example, when you see some channels and don't want to lose the link to it

**Comment by andrasfelfoldi** [comment]
This issue is still not implemented right?

I'd be happy to add it as an optional tab on the main page. It will take some time though as I'm not really experienced in Android develpoment so I will have to do some good old docs reading first.

**Comment by mhmdanas** [comment]
@andrasfelfoldi yes, this has not been implemented yet; if you are still interested feel free to make a comment and I'll assign you to this issue.

**Comment by ronidee** [comment]
@andrasfelfoldi I'd be happy to help if you don't have enough time.
I have worked with android in the past, but not an experienced android dev i would say.

**Comment by Juppy6** [comment]
From the guilty creator of the duplicate 7327 after many other duplicates I missed, thank you for getting this addition talked about again. 
I'm no developer but thought I'd just mention that the queue that is accessed from the pull down menu is already a perfect layout (mini player at bottom is nice with the queue) so if this can easily be put into an optional tab then... yay!
I would love to see added behaviour of - if a queue is already open and you were to press play on another video/song then the queue would not be lost but continued afterwards to avoid those lost queues. 
Thanks again you clever bunch!

**Comment by AbduAmeen** [comment]
This issue is a critical issue. I find that it could be easier to just incorporate the queue directly within the "Now playing" within another tab. 

It would be another tab within here which just renders the page. 
![image](https://user-images.githubusercontent.com/49756002/160264597-caa8d700-ddd9-42ce-9c9c-b8145ec22aa8.png)

This would be a first step with integrating the "Now playing" with the play queue, which I think is the correct way, considering that this standard practice.

**Comment by HybridAU** [comment]
Hey @SameenAhnaf that does seem like a good spot for the icon, but essentially the main reason I went with a button in the main menu rather than create a tab or add an icon in the top right was that it was the easiest change with the minimum amount of code required. 

I haven't done much work with Java so I'm not sure how hard it would be to add a button there, but you would need to be sure that it was only visible and active while the overlay showing the play button was active, you wouldn't want it to show over a video while it's playing, or be active (but not shown) and people accidentally open the play list by tapping on the video.

My gut feeling is that would be harder to implement and maintain.
